Tribler/Core/Libtorrent/LibtorrentMgr.py

```diff
@@ -631,9 +655,326 @@ def post_session_stats(self, hops=None):
         elif hasattr(self.ltsessions[hops], "post_session_stats"):
             self.ltsessions[hops].post_session_stats()
 
+    #####################################
+    # MOVED FROM LAUNCHMANY AND SESSION #
+    #####################################
 
-def encode_atp(atp):
-    for k, v in atp.items():
-        if isinstance(v, str):
-            atp[k] = v.encode('utf-8')
-    return atp
+    def add(self, tdef, config=None, delay=0, hidden=False, checkpoint_disabled=False):
+        infohash = tdef.get_infohash()
+
+        # Create the destination directory if it does not exist yet
+        try:
+            if not os.path.isdir(config.get_dest_dir()):
+                os.makedirs(config.get_dest_dir())
+        except OSError:
+            self._logger.error("Unable to create the download destination directory.")
+
+        if config.get_time_added() == 0:
+            config.set_time_added(int(timemod.time()))
+
+        # Check if running or saved on disk
+        if infohash in self.downloads:
+            self._logger.info("Torrent already exists in the downloads. Infohash:%s", hexlify(infohash))
+            return self.downloads[infohash]
+
+        download = LibtorrentDownloadImpl(self.tribler_session, tdef)
+
+        config = config or self.load_download_config_by_infohash(infohash)  # not already resuming
+
+        # Store in list of Downloads, always.
+        self.downloads[infohash] = download
+        download.setup(config, delay=delay, checkpoint_disabled=checkpoint_disabled,
+                       hidden=hidden or config.get_bootstrap_download())
+        return download
+
+    async def remove(self, download, removecontent=False, removestate=True):
```

Comment:
Why rename these keyword arguments. What's wrong with using underscores to separate words?

Reply:
Nothing wrong with the names. It was just easier to change `remove_state` for all the calls to `remove_download` method to become the same as `removestate` and `removecontent` arguments used by `metainfo_dl.stop`, than to the reverse.

Reply:
Renamed to use underscore